torch/csrc/autograd/variable.cpp

```diff
@@ -340,17 +343,51 @@ const std::shared_ptr<torch::autograd::Node>& VariableHooks::grad_fn(const Tenso
       // This is an indirect rebase_history due to another view or the base being modified inplace
       handle_view_on_rebase(diff_view_meta, /* indirect */ true);
       TORCH_INTERNAL_ASSERT(diff_view_meta->output_nr_ == 0);
-      auto fn = std::make_shared<torch::autograd::generated::AsStridedBackward>();
-      fn->self_geometry = at::TensorGeometry(diff_view_meta->base_);
-      fn->size = self.sizes().vec();
-      fn->stride = self.strides().vec();
-      fn->storage_offset = self.storage_offset();
-      fn->set_next_edges(torch::autograd::collect_next_edges(diff_view_meta->base_));
-      fn->add_input_metadata(
-        diff_view_meta->base_.options()
-      , self.sizes() // Note: sizes(), not base_.sizes(), is intentional
-      , diff_view_meta->base_.device());
-      diff_view_meta->grad_fn_ = std::move(fn);
+      // Note [View + Inplace update for view tensor]
+      // An inplace update happened on self Tensor (which is a view).
+      // For example:
+      //   view_1 = view_op_1(diff_view_meta->base_)
+      //   view_2 = view_op_2(view_1)
+      //   ...
+      //   self = view_op_n(view_n-1)
+      //   self = inplace_op(self)
+      //
+      // For CPU/CUDA backends, we employ one AsStridedBackward Node to represent the chain of
+      // view backward ops for effienciency.
+      //
+      // However in XLA backend we don't have full support of AsStridedBackward, we instead run a full
+      // forward pass with a tensor that requires gradient to get proper grad_fn setup,
+      // then save it to DifferentiableViewMeta for future use.
+      // This is fairly cheap for XLA lazy tensor approach(but really expensive for CPU/CUDA).
```

Comment:
nit: missing space before parenthesis.
nit: I would say "but would be really expensive" (as we don't actually do it for them).